packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/PdfViewerHighlight.tsx

```diff
@@ -54,40 +56,73 @@ const PdfViewerHighlight: FC<Props> = ({
 
   const { textDivs } = pdfRenderedText || {};
 
-  const highlightBoxes = useMemo(() => {
+  const highlightShapes = useMemo(() => {
     highlighter?.setTextContentDivs(textDivs);
     return highlights.map(highlight => {
       return highlighter?.getHighlight(highlight);
     });
   }, [highlighter, highlights, textDivs]);
 
+  const highlightDivRef = useRef<HTMLDivElement | null>(null);
+  useScrollIntoActiveHighlight(highlightDivRef, highlightShapes, activeIds);
+
+  return (
+    <div
+      ref={highlightDivRef}
+      className={cx(`${settings.prefix}--document-preview-pdf-viewer-highlight`, className)}
+    >
+      {highlightShapes.map((shape, hlIndex) => {
+        const active = shape?.highlightId ? activeIds?.includes(shape.highlightId) : false;
+        return (
+          <Highlight
+            key={shape?.highlightId || `highlight_${hlIndex}`}
```

Comment:
How often will the "else" case resolve here? We shouldn't use array index for the React `key` -- that can easily lead to bugs.

Reply:
The `shape.highlightId` is currently optional, but I refactor the code to make the `highlightId` required.